### ISSUE TITLE
Indicators drawn incorrectly after text-scale-adjust.

### ISSUE THREAD

**Issue by aaronbieber** [alpaker/fill-column-indicator#57]
After running `text-scale-adjust` once, increasing the font size somewhat, the fill column indicators don't move. With `fill-column` set to 80, after running `text-scale-adjust`, the indicators appear to be in column 70 (the same visual location where they were previously, but now the incorrect text column). The indicators on lines containing text are pushed out to approximately one space beyond the last character of the line and they do not align with one another.

Toggling `fci-mode` doesn't resolve the issue.

I'm using Emacs For OS X version 24.4.1 in the GUI.

**Comment by alpaker** [comment]
Thanks for the report.  This is a known issue, actually, and one that, as far as I can tell, would require non-trivial changes to Emacs's display engine to fix.  But I haven't thought about it for a while, so maybe with fresh eyes a solution will suggest itself to me.  I'll update here after looking again.

**Comment by fsnoon** [comment]
Alas, I'm often resizing text in my buffers.  Sort of makes this package useless to me.  A pity.

**Comment by fsnoon** [comment]
After investigating some more, I find that a `fci-rule-column` (or `fill-column`) of 92 will get me a margin line at column 80 when I've adjusted the text +1 bigger.  So I can also work around the problem by scaling the designated column number to fit my text adjustment: in practice there are usually just a couple of text sizes I use and a couple of column margins I care about, so I can figure out the magic numbers for what I'm doing.

**Comment by zw963** [comment]
If exist unicode chars (e.g. chinese), 92 is not a fixed number to deduce.

**Comment by zw963** [comment]
emacs 24.5, linux X86_64.  same problem.

**Comment by deb0ch** [comment]
I'm not sure, but aren't there new functions in emacs25 to get character width or text's displayed width ?

Could it help with this issue ?

**Comment by naclander** [comment]
A workaround for this is to use [zoom-frm](https://github.com/emacsmirror/zoom-frm) instead of adjusting font size. zoom-frm-in and zoom-frm-out ( kind of ) give you the same behavior.

My init.el contains:
```elisp
(if window-system (progn
    (global-set-key (kbd "C--" ) 'zoom-frm-out)
    (global-set-key (kbd "C-=") 'zoom-frm-in)))
````

**Comment by adamsmd** [comment]
It looks like pull request #80 fixes this issue.  (It is a replacement for pull request #76.)  Any chance that could be merged soon?

**Comment by naclander** [comment]
@alpaker can we merge #80?

**Comment by alpaker** [comment]
@naclander @adamsmd I seem to recall finding an issue with the approach the PR takes, but I can't recall precisely what.  (It's also entirely possible I'm imagining that.)  I'll take a look over the weekend.

**Comment by alpaker** [comment]
With the right combination of font dimension and text scaling factor, this approach can result in incorrect display:
![image](https://user-images.githubusercontent.com/764632/41204200-864e7f5a-6caf-11e8-9aa5-2b697cbe72e0.png)
Looking into the right tweak.

**Comment by rocian** [comment]
@alpaker I have just committed a patch on the same pull request (#80) that solve the incorrect display.

**Comment by alpaker** [comment]
Thanks.  I'll take another look tonight.

**Comment by dzhu** [comment]
The following appears to solve the positioning issue for me (only slightly tested):
```
(defconst fci-padding-display
  '((when (not (fci-competing-overlay-p buffer-position))
      . (space :align-to (fci-column . width)))   ; <-- instead of `:align-to fci-column`
    (space :width 0)))
```
If non-textual rules are being used and you increase the text size, you get a dashed line (each line contains only a piece with the original line height; that will need to be fixed separately), but at least it is at the right column. Also, I'm running Emacs compiled from git; I don't know when it became possible to do this.

Reference: https://www.gnu.org/software/emacs/manual/html_node/elisp/Pixel-Specification.html

**Comment by alpaker** [comment]
The functionality provided by this package has now been implemented natively in Emacs as `display-fill-column-indicator-mode`, available as of version 27.0.90. Unless you are forced to use an older Emacs, you should use the native implementation instead of `fci-mode`. The former is unaffected by most of the compatibility issues that this add-on has.
